pkg/cmd/cli/cmd/deploy.go

```diff
@@ -218,16 +246,58 @@ func (c *retryDeploymentCommand) retry(config *deployapi.DeploymentConfig, out i
 	return err
 }
 
+// cancelDeploymentCommand cancels the in-progress deployments.
+type cancelDeploymentCommand struct {
+	client deployCommandClient
+}
+
+// cancel cancels any deployment process in progress for config.
+func (c *cancelDeploymentCommand) cancel(config *deployapi.DeploymentConfig, out io.Writer) error {
+	deployments, err := c.client.ListDeploymentsForConfig(config.Namespace, config.Name)
+	if err != nil {
+		return err
+	}
+	for _, deployment := range deployments.Items {
+		status := deployutil.DeploymentStatusFor(&deployment)
+
+		switch status {
+		case deployapi.DeploymentStatusNew,
+			deployapi.DeploymentStatusPending,
+			deployapi.DeploymentStatusRunning:
+
+			if deployutil.IsDeploymentCancelled(&deployment) {
+				return nil
+			}
+
+			deployment.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
+			deployment.Annotations[deployapi.DeploymentStatusReasonAnnotation] = deployapi.DeploymentCancelledByUser
+			_, err := c.client.UpdateDeployment(&deployment)
+			if err == nil {
+				fmt.Fprintf(out, "cancelled #%d\n", config.LatestVersion)
```

Comment:
The deployment name is deterministic based on the config, and the deployer pod's name is deterministic based on the deployment. So, I'm not sure how the list will ever be greater than 1. Given that, why not just use `LatestDeploymentNameForConfig` and skip the list/loop?